dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/TrackedEntityAttributeObjectBundleHook.java

```diff
@@ -0,0 +1,55 @@
+package org.hisp.dhis.dxf2.metadata.objectbundle.hooks;
+
+import org.hisp.dhis.common.IdentifiableObject;
+import org.hisp.dhis.dxf2.metadata.objectbundle.ObjectBundle;
+import org.hisp.dhis.dxf2.utils.RenderingValidationUtils;
+import org.hisp.dhis.feedback.ErrorCode;
+import org.hisp.dhis.feedback.ErrorReport;
+import org.hisp.dhis.render.DeviceRenderTypeMap;
+import org.hisp.dhis.render.RenderDevice;
+import org.hisp.dhis.render.type.ValueTypeRenderingObject;
+import org.hisp.dhis.trackedentity.TrackedEntityAttribute;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TrackedEntityAttributeObjectBundleHook
+    extends AbstractObjectBundleHook
+{
+
+    @Override
+    public <T extends IdentifiableObject> List<ErrorReport> validate( T object, ObjectBundle bundle )
+    {
+        List<ErrorReport> errorReports = new ArrayList<>();
+
+        if ( object != null && object.getClass().isAssignableFrom( TrackedEntityAttribute.class ) )
+        {
+            TrackedEntityAttribute attr = (TrackedEntityAttribute) object;
+            DeviceRenderTypeMap<ValueTypeRenderingObject> map = attr.getRenderType();
+
+            if ( map != null )
```

Comment:
Maybe you can just check map == null here and break out? so we don't have another level of indent